web-app/src/resources/elements/panels/main-panel.js

```diff
@@ -267,6 +267,10 @@ export class MainPanel {
     selectedFileChanged() {
         this.clear();
         if (this.selectedFile) {
+
```

Comment:
my only comment here would be to avoid the empty whitespace above and below.  it doesn't aid reading.  I am not a huge fan of comments generally since the code should be clear in what it is doing or you should refractor to create methods (like resetSettingsOnLoad( ) etc.).  However I am ok with the comment.  I am not sure if I am fan since as I stated I am always opening large 600 dpi images.  So I have to constantly re-zoom out.  Maybe this need to be a user preference setting ("reset zoom to 100% file load")

Reply:
Let's do this.  Let's add the setting/preference and then in selectedFileChanged() add a "resetOnLoad()" method where we can conditionally check the perferences and reset the scaling factor.  We can then unit test that resetOnLoad function.  The current selectedFileChanged( ) does call filemanager which is always a bit tricky to test since it is actually a bridge object to the electron side.

Reply:
I will add a new "fit new image to screen" setting and a resetOnLoad() method.  I need to learn how to get the image size from the objectURL.

Reply:
I don't think you WANT fit new image to screen.  You want it to either remember the zoom factor OR reset to 100%.  I rarely am at 100%... I am usually at some factor of that... 25% or 50%.  but it varies so I would like it to retain what the last value was.  ie. I am not a fan of your improvement (to me it ads an extra step).  However I am willing to go with it if you can add a preference I can set to retain it.  The scaling factor then changes to 1 if it is to reset or remains as it is otherwise.

Reply:
Keep in mind, your scan was doing off some random set of images.  My scans are ALWAYS from a full vario stock sheet.  I scan the page, cover the holes and any stamps I don't want to process and proceed.  Here are the items I scanned today from my purchases from APS yesterday

![img021](https://user-images.githubusercontent.com/3221886/190003816-a2ed9bb6-888c-439a-80f2-c5f4218169e4.jpg)